dev-docs/RFCs/v1.0/react-map-gl-draw.md

```diff
@@ -0,0 +1,180 @@
+# RFC: react-map-gl-draw
+
+## Background
+
+react-map-gl currently does not support drawing functions. However, we have got a couple of [users](https://github.com/uber/react-map-gl/issues/725) interested in this capability. Also it is one of P0 features on Kepler.gl 2019 [roadmap](https://github.com/uber/kepler.gl/wiki/Kepler.gl-2019-Roadmap#allow-drawing-on-map-to-create-paths-and-polygons--). 
+
+Although [Mapbox/mapbox-gl-draw](https://github.com/mapbox/mapbox-gl-draw) provides drawing and editing features, because of its manipulating internal states, it cannot work well with React / Redux  framework and therefore cannot be integrated with `react-map-gl`.
+Another `vis.gl` framework [Nebula.gl](http://nebula.gl) also provides geo editing functionality, but it heavily depends on [`deck.gl`](https://deck.gl/), which may not be suitable for non `deck.gl` use cases.
+
+## Proposal
+
+`react-map-gl-draw` is a react based geo editing library, providing drawing primitives, specifically `points`, `lines`, `polygons`, and `rectangles`.
+
+## Options 
+- `MapContextProvider` (React.Component, required) - A [Context.Provider](https://reactjs.org/docs/context.html#contextprovider) component. If supplied, will be rendered as the ancestor to all children. The passed through context contains the following values:
+  - `viewport` ([Viewport](/docs/api-reference/viewport.md)) - the current viewport
+  - `eventManager` ([EventManager](https://github.com/uber-web/mjolnir.js/blob/master/docs/api-reference/event-manager.md))
+
+- `mode` (String, Optional) - `react-map-gl` is stateless, user has complete control of the `mode`.
+  - `EditorModes.READ_ONLY` - Not interactive. This is the default mode.
+  - `EditorModes.SELECT_FEATURE` - Lets you select, delete, and drag features.
+  - `EditorModes.EDIT_VERTEX` - Lets you select, delete, and drag vertices; and drag features.
```

Comment:
I'd prefer `selectable` and `editable` be independent flags from the modes. This would support the scenario where the app allows the user to draw new features and edit existing features at the same time.

Reply:
What would it take to get these `EditorModes` to just be nebula.gl `EditMode`s?

Reply:
This refactor is the first step, which help me to know better what kinds of behaviors are expected under different modes. Then the next step is to extract handlers for different modes by extending edit-modes module.